logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogStreamBatchReaderTest.java

```diff
@@ -0,0 +1,335 @@
+
+    assertThat(batch1.hasNext()).isTrue();
+    assertThat(batch1.next().getPosition()).isEqualTo(eventPosition2);
+
+    assertThat(batch1.hasNext()).isFalse();
+
+    final var batch2 = batchReader.next();
+    assertThat(batch2.hasNext()).isTrue();
+    assertThat(batch2.next().getPosition()).isEqualTo(eventPosition3);
+  }
+
+  @Test
+  public void shouldSkipEventsInBatch() {
+    // given
+    final long eventPosition1 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition3 = writerRule.sourceEventPosition(2L).writeEvent(EVENT_VALUE);
+
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch1 = batchReader.next();
+    assertThat(batch1.hasNext()).isTrue();
+    assertThat(batch1.next().getPosition()).isEqualTo(eventPosition1);
+
+    // when
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch2 = batchReader.next();
+    assertThat(batch2.hasNext()).isTrue();
+    assertThat(batch2.next().getPosition()).isEqualTo(eventPosition3);
+  }
+
+  @Test
+  public void shouldSeekToHeadIfNegative() {
+    // given
+    final long eventPosition1 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(-1L);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch = batchReader.next();
+    assertThat(batch.hasNext()).isTrue();
+    assertThat(batch.next().getPosition()).isEqualTo(eventPosition1);
+  }
+
+  @Test
+  public void shouldSeekToNextBatch() {
+    // given
+    writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition2 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition3 = writerRule.sourceEventPosition(2L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(eventPosition2);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch = batchReader.next();
+    assertThat(batch.hasNext()).isTrue();
+    assertThat(batch.next().getPosition()).isEqualTo(eventPosition3);
+  }
+
+  @Test
+  public void shouldSeekToNextEventWithinBatch() {
+    // given
+    final long eventPosition1 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition2 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    writerRule.sourceEventPosition(2L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(eventPosition1);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch = batchReader.next();
+    assertThat(batch.hasNext()).isTrue();
+    assertThat(batch.next().getPosition()).isEqualTo(eventPosition2);
+  }
```

Comment:
hm this is a bit confusing :thinking:

Reply:
Please explain what do you think is confusing. Do you have a suggestion on how to make it less confusing?

Reply:
Sorry. The method is called seekToNextBatch, I would expect it will go to the next batch like it does on the other test method OR to the begin of this one, but not in the middle this is confusing and probably error prone, since we miss here then one event?

Reply:
I see this point.

But pragmatically, it will no be used this way. The batch reader will only be used (at the moment) in the replay state machine. There, we set the position only after a batch is read completely. So, we will pass always the last event position of the batch in this method.

Under this assumption, the additional logic to seek to the beginning of the batch or the next one will not be used. So, I favor not implementing it. 

Do you agree?

Reply:
The point is "at the moment". You rely on the implementation instead of the api. I would like that we define the API as it should be and then implement it and expect also that it works like it is defined. If you dont want to implement it,  then you should make at least a note at the java doc, otherwise it is a trap we fall into later.

Reply:
Okay. I'll change the API to seek to the next batch. This should make it less confusing :wink:

Reply:
Thanks :heart: